packages/kbn-config-schema/src/types/map_type.ts

```diff
@@ -53,7 +53,9 @@ export class MapOfType<K, V> extends Type<Map<K, V>> {
         const childPathWithIndex = reason.path.slice();
         childPathWithIndex.splice(path.length, 0, entryKey.toString());
 
-        return new SchemaTypeError(reason.message, childPathWithIndex);
+        return reason instanceof SchemaTypesError
```

Comment:
__note:__ while working on this PR I've noticed that we miss "composite" errors in `MapOfTime`